src/app/features/charts/components/median-barchart/median-barchart.component.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { MedianExpression } from '../../../../models/genes';
+
+// -------------------------------------------------------------------------- //
+// Tests
+// -------------------------------------------------------------------------- //
+const XAXIS_LABEL = 'BRAIN REGION';
+const YAXIS_LABEL = 'LOG2CPM';
+const FULL_MOCK_DATA = geneMock1.medianexpression;
+const TISSUES = ['TCX', 'PHG', 'STG'];
+const SMALL_MOCK_DATA = [
+  {
+    ensembl_gene_id: 'MOCK_GENE',
+    minimumlogcpm: -2.54173091337051,
+    quartile1logcpm: -1.03358030635935,
+    medianlogcpm: 0.483801733963266,
+    meanlogcpm: -0.517398199667356,
+    quartile3logcpm: -0.0800759845652829,
+    maximumlogcpm: 2.32290808871289,
+    tissue: TISSUES[0],
+  },
+  {
+    ensembl_gene_id: 'MOCK_GENE',
+    minimumlogcpm: -2.44077907413711,
+    quartile1logcpm: -0.592671867557559,
+    medianlogcpm: 0.013739530502129,
+    meanlogcpm: -0.0324143336865,
+    quartile3logcpm: 0.49577213202412,
+    maximumlogcpm: 2.23019575245731,
+    tissue: TISSUES[1],
+  },
+  {
+    ensembl_gene_id: 'MOCK_GENE',
+    minimumlogcpm: -5.03189866356294,
+    quartile1logcpm: -1.02644563959975,
+    medianlogcpm: 0.176348063122062,
+    meanlogcpm: -0.323038107200895,
+    quartile3logcpm: 0.391874711168331,
+    maximumlogcpm: 1.9113258251877,
+    tissue: TISSUES[2],
+  },
+];
+
+describe('Component: BarChart - Median', () => {
+  let fixture: ComponentFixture<MedianBarChartComponent>;
+  let component: MedianBarChartComponent;
+  let element: HTMLElement;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [MedianBarChartComponent],
+      imports: [RouterTestingModule],
+      providers: [HelperService],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MedianBarChartComponent);
+    component = fixture.componentInstance;
+  });
+
+  const setUp = (
+    data: MedianExpression[] = FULL_MOCK_DATA,
+    xAxisLabel: string = XAXIS_LABEL,
+    yAxisLabel: string = YAXIS_LABEL
+  ) => {
+    component.data = data;
+    component.xAxisLabel = xAxisLabel;
+    component.yAxisLabel = yAxisLabel;
+
+    fixture.detectChanges();
+    element = fixture.nativeElement;
+    const chart = element.querySelector('svg g');
+    return { chart };
+  };
+
+  it('should create', () => {
+    setUp();
+    expect(component).toBeTruthy();
+  });
+
+  it('should not render the chart if there is no data', () => {
+    const { chart } = setUp([]);
+
+    expect(component.data?.length).toEqual(0);
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should not render the chart if all values are negative', () => {
+    const { chart } = setUp(
+      SMALL_MOCK_DATA.map((obj) => {
+        return { ...obj, medianlogcpm: -1 * obj.medianlogcpm };
+      })
+    );
+
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
```

Comment:
Match [the current behavior](https://agora-develop.adknowledgeportal.org/genes/ENSG00000228411/evidence/rna), which does not render the chart when all medianlogcpm values are negative